packages/admin/admin-generator/src/commands/generate/generateForm/__tests__/staticSelect.ts

```diff
@@ -0,0 +1,265 @@
+import { ESLint } from "eslint";
+import { promises as fs } from "fs";
+import { buildSchema, introspectionFromSchema } from "graphql";
+import { type JsxAttribute, Project, SyntaxKind } from "ts-morph";
+
+import { type FormConfig } from "../../generate-command";
+import { configsFromSourceFile } from "../../utils/tsMorphHelper";
+import { generateForm } from "../generateForm";
+
+const project = new Project({
+    tsConfigFilePath: "src/.test/tsconfig.json",
+    skipAddingFilesFromTsConfig: true,
+});
+
+async function lintCode(contents: string, filePath: string) {
+    const eslint = new ESLint({
+        cwd: process.cwd(),
+        fix: true,
+    });
+
+    await fs.writeFile(`src/.test/${filePath}`, contents);
+    const lintResult = await eslint.lintText(contents, {
+        filePath: `src/.test/${filePath}`,
+    });
+    if (lintResult[0].errorCount > 0 || lintResult[0].fatalErrorCount > 0) {
+        const errorMessage = lintResult[0].messages
+            .map((message) => {
+                return message.message;
+            })
+            .join(".");
+
+        throw new Error(`Linting error: \n${errorMessage}`);
+    } else if (lintResult[0].output != null) {
+        const tsSource = project.createSourceFile(`src/.test/${filePath}`, lintResult[0].output, { overwrite: true });
+        return tsSource;
+    } else {
+        throw new Error("No output from linting");
+    }
+}
+
+function parseString(sourceFileText: string) {
+    const tsSource = project.createSourceFile("test.tsx", sourceFileText, { overwrite: true });
+    return configsFromSourceFile(tsSource);
+}
+
+describe("Form StaticSelect", () => {
+    const schema = buildSchema(`
+        enum ProductType {
+            Cap
+            Shirt
+            Tie
+        }
+        type Query {
+            product(id: ID!): Product!
+            products: [Product!]!
+        }
+        type Mutation {
+            createProduct(title: String!): Product!
+        }
+        type Product {
+            id: ID!
+            title: String!
+            type: ProductType!
+        }
+    `);
+
+    const gqlIntrospection = introspectionFromSchema(schema);
+
+    it("creates RadioGroupField with values inferred from gql schema", async () => {
+        const config = parseString(`
+            import { FormConfig } from "@comet/admin-generator";
+            import { GQLProduct } from "@src/graphql.generated";
+
+            export const ProductForm: FormConfig<GQLProduct> = {
+                type: "form",
+                gqlType: "Product",
+                fields: [
+                    {
+                        type: "staticSelect",
+                        name: "type",
+                    },
+                ],
+            }
+        `);
```

Comment:
Writing the generator config as strings in the tests could result in hard-to-debug errors. Could we do this differently?

Reply:
yes, but then each test needs an additional file, wich makes the tests much harder to read and write.

But you are totally right, without proper ts checks you can easily write invalid code

Reply:
We could run tsc (during running test). That still gives no IDE support though.

Reply:
Let's keep it like this for now. We can come up with a better solution in the future.